prediction/src/algorithms/segment/src/data_generation.py

```diff
@@ -0,0 +1,61 @@
+import glob
+import os
+
+import numpy as np
+import pylidc as pl
+
+
+def get_dicom_paths(in_docker=True):
+    """Return DICOM paths to all LIDC directories
+    e.g. ['../images_full/LIDC-IDRI-0001/1.3.6.1.4.1.14519.5.2.1.6279.6001.298806137288633453246975630178/' \
+          '1.3.6.1.4.1.14519.5.2.1.6279.6001.179049373636438705059720603192']"""
+    if in_docker:
+        return glob.glob("../images_full/LIDC-IDRI-*/**/**")
+    else:
+        return glob.glob("../tests/assets/test_image_data/full/LIDC-IDRI-*/**/**")
+
+
+def prepare_training_data():
+    current_dir = os.path.dirname(os.path.realpath(__file__))
+    assets_dir = os.path.abspath(os.path.join(current_dir, '../assets'))
+
+    dicom_paths = sorted(get_dicom_paths())
+    for path in dicom_paths:
+        directories = path.split('/')
+        lidc_id = directories[2]
+        lung_patient_file = os.path.join(assets_dir, "segmented_lung_patient_{}".format(lidc_id))
+
+        if os.path.isfile(lung_patient_file):
+            continue
+
+        # Compute and save binary mask with information whether pixel is cancerous
+        scan = pl.query(pl.Scan).filter(pl.Scan.patient_id == lidc_id).first()
+        if scan is None:
+            print("Scan for path '{}' was not found".format(path))
+            continue
+        vol = scan.to_volume(verbose=False)
```

Comment:
In order to get the volume of a patient scan, the leading zeros of the dicom file names had to be removed.